src/vector/_pytree.py

```diff
@@ -0,0 +1,202 @@
+from __future__ import annotations
+
+from functools import partial
+from typing import TYPE_CHECKING, Any
+
+import numpy
+
+if TYPE_CHECKING:
+    from optree.pytree import ReexportedPyTreeModule
+
+from vector._methods import (
+    Vector2D,
+    Vector3D,
+    Vector4D,
+)
+from vector.backends.numpy import (
+    MomentumNumpy2D,
+    MomentumNumpy3D,
+    MomentumNumpy4D,
+    VectorNumpy,
+    VectorNumpy2D,
+    VectorNumpy3D,
+    VectorNumpy4D,
+)
+from vector.backends.object import (
+    MomentumObject2D,
+    MomentumObject3D,
+    MomentumObject4D,
+    VectorObject2D,
+    VectorObject3D,
+    VectorObject4D,
+)
+
+Children = tuple[Any, ...]
+MetaData = tuple[type, ...]
```

Comment:
FYI: the Python 3.8 job is failing because this syntax is not supported on Python 3.8. Maybe we can use a Python 3.8 friendly syntax for this PR and worry about removing 3.8 support in another issue/PR.